Code/GraphMol/SubstructLibrary/SubstructLibrary.h

```diff
@@ -213,10 +218,13 @@ class RDKIT_SUBSTRUCTLIBRARY_EXPORT CachedTrustedSmilesMolHolder
     return size() - 1;
   }
 
-  virtual boost::shared_ptr<ROMol> getMol(unsigned int idx) const {
+  virtual boost::shared_ptr<ROMol> getMol(unsigned int idx, bool sanitize=true) const {
     if (idx >= mols.size()) throw IndexErrorException(idx);
 
     RWMol *m = SmilesToMol(mols[idx], 0, false);
+    if (sanitize) { // only need ring info for sanitiation
```

Comment:
```suggestion
    if (sanitize) { // only need ring info for sanitization
```

Reply:
It shouldn't be the default behavior.  When the user calls getMol, they do get a sanitized molecule.  Internally, sanitize is only set to true for a ring query, or at least that is the intention.